src/progress_bar.rs

```diff
@@ -607,10 +607,12 @@ impl MultiProgress {
         match &pb.state.lock().unwrap().draw_target.kind {
             ProgressDrawTargetKind::Term { .. } => {}
             ProgressDrawTargetKind::Remote { idx, .. } => {
-                state.objects[*idx] = None;
-                state.free_set.insert(*idx);
-                if let Some(idx) = state.ordering.iter().position(|x| *x == *idx) {
-                    state.ordering.remove(idx);
+                if state.objects[*idx].is_some() {
```

Comment:
Let's use `take()` here.

Reply:
I'm not sure I understand. `take`  returns us `Option<MultiObject>`, which we don't need.